gnovm/cmd/gno/lint.go

```diff
@@ -140,6 +164,67 @@ func execLint(cfg *lintCfg, args []string, io commands.IO) error {
 	return nil
 }
 
+func lintTypeCheck(io commands.IO, memPkg *std.MemPackage, testStore gno.Store) (errorsFound bool, err error) {
+	tcErr := gno.TypeCheckMemPackageTest(memPkg, testStore)
+	if tcErr == nil {
+		return false, nil
+	}
+
+	errs := multierr.Errors(tcErr)
+	for _, err := range errs {
+		switch err := err.(type) {
+		case types.Error:
+			io.ErrPrintln(lintIssue{
+				Code:       lintTypeCheckError,
+				Msg:        err.Msg,
+				Confidence: 1,
+				Location:   err.Fset.Position(err.Pos).String(),
+			})
+		case scanner.ErrorList:
+			for _, scErr := range err {
+				io.ErrPrintln(lintIssue{
+					Code:       lintParserError,
+					Msg:        scErr.Msg,
+					Confidence: 1,
+					Location:   scErr.Pos.String(),
+				})
+			}
+		case scanner.Error:
+			io.ErrPrintln(lintIssue{
+				Code:       lintParserError,
+				Msg:        err.Msg,
+				Confidence: 1,
+				Location:   err.Pos.String(),
+			})
+		default:
+			//nolint
+			return false, fmt.Errorf("unexpected error type: %T", err)
+		}
+	}
+	return true, nil
```

Comment:
what is the situation that errorsFound is true while `err` is nil?

Reply:
None, really: further above, we check that `tcErr == nil`. So, at this point, there will always be at least an error that was printed.